handlers/common.go

```diff
@@ -177,6 +124,10 @@ func GetValidatorEarnings(validators []uint64, currency string) (*types.Validato
 	earnings7d := income.ClIncome7d + income.ElIncome7d
 	earnings31d := income.ClIncome31d + income.ElIncome31d
 
+	if totalDeposits == 0 {
+		totalDeposits = utils.Config.Chain.Config.MaxEffectiveBalance
+	}
```

Comment:
I know this block was "here" before, but please let us ponder on it briefly: What exactly does that do?

**1)** I believe this is solely done to avoid divisions by 0 below. What do you think?

**2)** This should result in wrong results whenever `func GetValidatorEarnings` is called for more than one validator. Should it not be something like `totalDeposits = utils.Config.Chain.Config.MaxEffectiveBalance * uint64(len(validators))`?
This should be save as `MaxEffectiveBalance` is set denominated in Gwei, see https://github.com/gobitfly/eth2-beaconchain-explorer/blob/26be5543a82326de3c52c5fc5a238e2bc7208df2/config/prater.chain.yml#L196
**3)** Have you considered putting this block into `func GetTotalValidatorDeposits`. I am not suggesting to do so, I am just curious what your POV is on that.

Reply:
adapted your` 2)` suggestion.
I would not do `3)` as I don't want to put to much business logic in a simple function that gets data from the db.

Reply:
Regarding 2), you used `+` instead of `*` here: https://github.com/gobitfly/eth2-beaconchain-explorer/pull/2190/files/f580b3eed31943b328bcae991e25cf6002d2bc9c..983e326b7714e1d86577e00eef16eb3a937452db#diff-19b5d68076eccab6d3c22ec9de45d10c78540bbbbadb54633312a83cabe56633R131